rust/connlib/tunnel/src/client.rs

```diff
@@ -93,7 +93,7 @@ where
     // FIXME: this cleanup connection is wrong!
     pub fn cleanup_connection(&self, id: ResourceId) {
         self.role_state.lock().on_connection_failed(id);
-        self.peer_connections.lock().remove(&id.into());
+        // self.peer_connections.lock().remove(&id.into());
```

Comment:
shouldn't we still cleanup the `peer_connections` state? or are we sure this is always called before it's inserted?

Reply:
I don't know to be honest. I think the connection cleanup isn't very consistent at the moment.

I am working on a few more refactorings which should make it a lot easier to do this consistently :)

I'd suggest deferring until then!